### PR TITLE
Allow more ast lines while trying to keep out non-user-source

### LINKED ISSUE #3849: [BUG]: ast viewer truncates the output in some cases
### Describe the bug

Given the following example: https://godbolt.org/z/GKza5dvrh

The Ast Viewer truncates the output but if I use `-Xclang -ast-dump` directly I do see the whole AST dump. I show both cases in the link above

### Steps to reproduce

Plugging the following C++ code into compiler explorer:

```cpp
void f();

 char Bar(auto ) {
    return {};
}

int main() {
    char c = Bar([]{});
    return c;
}
```

The AST Viewer does not show the whole AST for this code.

### Expected behavior

Showing the AST similar to what I see when using `-Xclang -ast-dump` 

### Reproduction link

https://godbolt.org/z/GKza5dvrh

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by partouf**
Fixes #3849 

The culprit for this issue is the `<invalid sloc>` notation for every top-level entry except for `f 'void ()'`, this caused the lines to be filtered out.

But main() also doesn't contain any source reference, so that's removed as well.

I added code to allow those to be unfiltered, but that caused some other unittest to fail (when including ctime) because of all the extra includes to be visible. So I subsequently tried to filter out more headers that don't really need to be visible (libc headers, linux headers and libc++/stdlib headers)

### CODE REVIEW

File: lib/llvm-ast.js [hunk 1]
```diff
@@ -117,9 +128,13 @@ export class LlvmAstParser {
                     // This is a system header or implicit definition,
                     // remove everything up to the next top level decl
                     // Top level decls with invalid sloc as the file don't change the most recent file
-                    const slocRegex = /<<invalid sloc>>/;
-                    if (!slocRegex.test(output[i].text)) {
+                    if (systemSource.test(output[i].text)) {
+                        // skip ast from this source
+                    } else if (userSource.test(output[i].text)) {
+                        continue;
```
Comment (RubenRBS) [inline lib/llvm-ast.js:134]: codecov complains that no test exercises this path, but I expected at least some lines would, no?

File: lib/llvm-ast.js [hunk 2]
```diff
@@ -103,6 +103,17 @@ export class LlvmAstParser {
         // Refers to the user's source file rather than a system header
         const sourceRegex = /<source>/g;
 
+        const slocRegex = /<<invalid sloc>>/;
+
+        // <<invalid sloc>, /app/hell.hpp:5:1>
```
Comment (RubenRBS) [inline lib/llvm-ast.js:108]: 🔥

**Review by RubenRBS — APPROVED** [review]

File: lib/llvm-ast.js (hunk 1 above)
Comment (partouf) [reply lib/llvm-ast.js:134]: that might be so, I probably only tested it in the browser, will see if I can provide the test

**Review by matsjla — APPROVED** [review]

**Comment by mattgodbolt** [comment]
Now live! thanks:)

**Pull request merged**